contracts/BIFI/strategies/Arbidex/StrategyArbidexMultiRewardsLP.sol

```diff
@@ -0,0 +1,338 @@
+
+        IMasterChef(chef).emergencyWithdraw(poolId);
+
+        uint256 wantBal = IERC20(want).balanceOf(address(this));
+        IERC20(want).transfer(vault, wantBal);
+    }
+
+    // pauses deposits and withdraws all funds from third party systems.
+    function panic() public onlyManager {
+        pause();
+        IMasterChef(chef).emergencyWithdraw(poolId);
+    }
+
+    function pause() public onlyManager {
+        _pause();
+
+        _removeAllowances();
+    }
+
+    function unpause() external onlyManager {
+        _unpause();
+
+        _giveAllowances();
+
+        deposit();
+    }
+
+    function _giveAllowances() internal {
+        IERC20(want).safeApprove(chef, type(uint).max);
+        IERC20(output).safeApprove(unirouter, type(uint).max);
+        IERC20(native).safeApprove(unirouter, type(uint).max);
+
+        IERC20(lpToken0).safeApprove(unirouter, 0);
+        IERC20(lpToken0).safeApprove(unirouter, type(uint).max);
+
+        IERC20(lpToken1).safeApprove(unirouter, 0);
+        IERC20(lpToken1).safeApprove(unirouter, type(uint).max);
+
+        if (rewardToNativeRoute.length != 0) {
+            for (uint i; i < rewardToNativeRoute.length; i++) {
+                IERC20(rewardToNativeRoute[i][0]).safeApprove(unirouter, 0);
+                IERC20(rewardToNativeRoute[i][0]).safeApprove(unirouter, type(uint).max);
+            }
+        }
+    }
+
+    function _removeAllowances() internal {
+        IERC20(want).safeApprove(chef, 0);
+        IERC20(output).safeApprove(unirouter, 0);
+        IERC20(native).safeApprove(unirouter, 0);
+
+        IERC20(lpToken0).safeApprove(unirouter, 0);
+        IERC20(lpToken1).safeApprove(unirouter, 0);
+
+        if (rewardToNativeRoute.length != 0) {
+            for (uint i; i < rewardToNativeRoute.length; i++) {
+                IERC20(rewardToNativeRoute[i][0]).safeApprove(unirouter, 0);
+            }
+        }
+    }
+
+    function addRewardRoute(address[] memory _rewardToNativeRoute) external onlyOwner {
+        IERC20(_rewardToNativeRoute[0]).safeApprove(unirouter, 0);
+        IERC20(_rewardToNativeRoute[0]).safeApprove(unirouter, type(uint).max);
+        rewardToNativeRoute.push(_rewardToNativeRoute);
+    }
+
+    function removeLastRewardRoute() external onlyManager {
+        address reward = rewardToNativeRoute[rewardToNativeRoute.length - 1][0];
+        if (reward != lpToken0 && reward != lpToken1) {
+            IERC20(reward).safeApprove(unirouter, 0);
+        }
+        rewardToNativeRoute.pop();
+    }
+
+    function outputToNative() external view returns (address[] memory) {
+        return outputToNativeRoute;
+    }
+
+    function nativeToLp0() external view returns (address[] memory) {
+        return nativeToLp0Route;
+    }
+
+    function nativeToLp1() external view returns (address[] memory) {
+        return nativeToLp1Route;
+    }
+
+    function rewardToNative() external view returns (address[][] memory) {
```

Comment:
this would be better with 
```
rewardToNative(uint i) external view returns (address[] memory) {
  return rewardToNativeRoute[i];
}
```
so its usable from explorer instead of 
<img width="392" alt="image" src="https://user-images.githubusercontent.com/71946691/232753450-6854d3fe-2e87-411b-89eb-22a356dc8510.png">
but then we need to also have `rewardLength() returns rewardToNativeRoute.length`.
or just add rewardToNative(uint i) as additional function and keep current one

Reply:
![image](https://user-images.githubusercontent.com/39047198/233789433-ed75a8ae-a377-4180-9746-cc11608bfa72.png)
The public variable rewardToNativeRoute has an index-based read option automatically on arbiscan (see screenshot). Do we really need another read function for this?

Reply:
https://arbiscan.io/address/0x632585cb1BC7f7f824E8d9777b6707f7396768F6#readContract
This is the deployed contract I took the screenshot from.

Reply:
Fair point, it's better and more convenient that way. I added the index-based view function as well as `rewardLength`.